libs/shared/rank/src/lib/shared-rank.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Grid, Box } from '@material-ui/core';
+import { Rating } from '@material-ui/lab';
+
+import './shared-rank.scss';
+
+export interface SharedRankProps {
+  rank: number;
+}
+
+export const SharedRank = (props: SharedRankProps) => {
+  const [rank, setRank] = React.useState(props.rank);
```

Comment:
Don't use " React.useState(props.rank); " use " useState(props.rank);"  and useState should have default value, from props you can initialise only with useEffect